.github/actions/bumpVersion/bumpVersion.js

```diff
@@ -47,69 +47,61 @@ let shouldRetry;
-            }
+        let newVersion;
 
-            // Maximum retries reached, fail this action.
-            core.setFailed(err);
+        // Fetch tags
+        console.log('Fetching tags from github...');
+        const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
+        return octokit.repos.listTags({
+            owner: repoOwner,
+            repo: repoName,
         })
-        .then(({currentPatchVersion, githubResponse}) => {
-            // Find the highest build version git tag
-            const tags = githubResponse.data.map(tag => tag.name);
-            console.log('Tags: ', tags);
-            const highestBuildNumber = Math.max(
-                ...(tags
-                    .filter(tag => tag.startsWith(currentPatchVersion))
-                    .map(tag => tag.split('-')[1])
-                ),
-            );
-            console.log('Highest build number from current patch version:', highestBuildNumber);
+            .catch(githubError => core.setFailed(githubError))
+            .then((githubResponse) => {
+                // Find the highest build version git tag
+                const tags = githubResponse.data.map(tag => tag.name);
+                console.log('Tags: ', tags);
+                const highestBuildNumber = Math.max(
+                    ...(tags
+                        .filter(tag => tag.startsWith(currentPatchVersion))
+                        .map(tag => tag.split('-')[1])
+                    ),
+                );
+                console.log('Highest build number from current patch version:', highestBuildNumber);
 
-            // Bump the build number again
-            const newBuildNumber = `${currentPatchVersion}-${highestBuildNumber + 1}`;
-            console.log(`Setting npm version for this PR to ${newBuildNumber}`);
-            return exec(`npm version ${newBuildNumber} -m "Update version to ${newBuildNumber}"`);
-        })
-        // eslint-disable-next-line no-loop-func
-        .then(({stdout}) => {
-            // NPM version successfully updated, update native versions - don't retry.
-            postVersionUpdateNative(stdout);
-            shouldRetry = false;
-        })
-        .catch(({stdout, stderr}) => {
-            // Log errors and retry
-            console.log(stdout);
-            console.error(stderr);
-        });
+                // Bump the build number again
+                newVersion = `${currentPatchVersion}-${highestBuildNumber + 1}`;
+                console.log(`Setting npm version for this PR to ${newVersion}`);
+                return exec(`npm version ${newVersion} -m "Update version to ${newVersion}"`);
+            })
+            // eslint-disable-next-line no-loop-func
+            .then(({stdout}) => {
+                // NPM version successfully updated, update native versions - don't retry.
+                console.log(stdout);
+                postVersionUpdateNative(newVersion);
```

Comment:
Do we need `const cleanNewVersion = semverClean(newVersion);` anymore now that this is a known semver version sent?

Reply:
That's a good question. I think that we do still need it because it will take `v1.0.1-8` and clean it to be `1.0.1-8`. I tested, and the `npm version` command is robust enough that it will strip the leading `v` on its own. But it also seems like it will automatically tack that `v` prefix onto the Git tag it creates.

Reply:
So in conclusion, we could change things so that we don't need it, but I also don't really see it introducing any complexity. Feel free to let me know if you disagree.